gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2010,8 +2007,30 @@ private Bucket getBucket(String bucketName) throws IOException {
       if (errorExtractor.itemNotFound(e)) {
         logger.atFiner().withCause(e).log("getBucket(%s): not found", bucketName);
         return null;
+      } else {
```

Comment:
Even if item was not found, request was sent.

Reply:
Updated with new implementation